pkg/apis/application/v1alpha1/types.go

```diff
@@ -437,6 +437,16 @@ type SyncPolicyAutomated struct {
 	Prune bool `json:"prune,omitempty" protobuf:"bytes,1,opt,name=prune"`
 	// SelfHeal enables auto-syncing if  (default: false)
 	SelfHeal bool `json:"selfHeal,omitempty" protobuf:"bytes,2,opt,name=selfHeal"`
+	// MaintenanceWindows disables auto-syncing during the configured time windows  (default: false)
+	MaintenanceWindows []*MaintenanceWindow `json:"maintenanceWindows,omitempty" protobuf:"bytes,3,opt,name=maintenanceWindows"`
```

Comment:
I love to create `type MaintenanceWindows = []*MaintenanceWindow` as you can then create func for it.

Reply:
I have done this in the new pull request.